robot/web/stream/server.go

```diff
@@ -319,6 +329,104 @@ func (server *Server) RemoveStream(ctx context.Context, req *streampb.RemoveStre
 	return &streampb.RemoveStreamResponse{}, nil
 }
 
+// GetStreamOptions implements part of the StreamServiceServer. It returns the available dynamic resolutions
+// for a given stream name. The resolutions are scaled down from the original resolution in the camera
+// properties.
+func (server *Server) GetStreamOptions(
+	ctx context.Context,
+	req *streampb.GetStreamOptionsRequest,
+) (*streampb.GetStreamOptionsResponse, error) {
+	if req.Name == "" {
+		return nil, errors.New("stream name is required")
+	}
+	cam, err := camera.FromRobot(server.robot, req.Name)
+	if err != nil {
+		return nil, err
+	}
+	// If the camera properties do not have intrinsic parameters,
+	// we need to sample a frame to get the width and height.
+	var width, height int
+	camProps, err := cam.Properties(ctx)
+	if err != nil || camProps.IntrinsicParams == nil || camProps.IntrinsicParams.Width == 0 || camProps.IntrinsicParams.Height == 0 {
```

Comment:
if the `err != nil` || case, can we log it somewhere (before it's overwritten by the new error from `sampleFrameSize`)